Add support for mbbiDirect and mbboDirect record types

JJL772 opened the pull request:
Initially I was going to add support for mbbi and mbbo, but the direct record types seem to be far more useful for our applications at SLAC.

Most of the changes here are in the record templates, you'll probably want to review commit-by-commit. 

Relies on #32 getting merged first.

Closes #33

JJL772 committed:
ENH: cleanup scripts

JJL772 committed:
ENH: support for mbboDirect and mbbiDirect record types

JJL772 committed:
ENH: mbbiDirect/mbboDirect record templates

JJL772 committed:
FIX: get building with C++03 again

JJL772 committed:
STY: apply clang-format

klauer commented on ek9000App/src/scripts/generate.py:
```diff
@@ -133,10 +135,16 @@ def set_default_bi(self):
         self.vals['ZNAM'] = 'low'
         self.vals['ONAM'] = 'high'
-        self.vals['PINI'] = 'YES'  # TODO: this seems wrong
+        self.vals['PINI'] = 'YES'
```
I think `PINI` configured to true for an output record may be in error. Am I misunderstanding? (or is my record behavior faulty? It's been a while since I've done much...)

As I read it, at startup of any given IOC, it will attempt to write the current value (0) for all output records using EK9K. That is to say, all digital outputs would be instructed to go to 0.

JJL772 replied:
Right, okay. I interpreted the comment as "YES seems like an invalid choice"

I'll take a look at the docs then

JJL772 replied:
I think your analysis is correct here. I suspect this hasn't been reported as an issue because most or all of our (admittedly few) ek9k deployments have the watchdog timer enabled with set-to-zero, so outputs get set to 0 regardless when restarting the IOC.

klauer commented on ek9000App/src/scripts/generate.py:
```diff
@@ -133,10 +135,16 @@ def set_default_bi(self):
+    def set_default_mbbi(self):
+        self.vals['SCAN'] = 'I/O Intr'
+        
+    def set_default_mbbo(self):
```
Not related to this line - for outputs, it'd be nice to have both setpoint and readback records.
Setpoint for what the user intends to write and readback for what the PLC currently reports.

For all of our recent EPICS IOCs (well, at least auto-generated ones) we've followed the AreaDetector convention of having records named as such:
```
pvname       <- the setpoint value
pvname_RBV   <- the corresponding readback value
```

It shouldn't be too difficult to add that record here, I'd think.

If you think it's a reasonable thing to add, let's put it out to another PR though - this one's big enough as is!

klauer commented on ek9000App/src/devEK9000.cpp:
```diff
@@ -66,7 +66,7 @@ std::list<devEK9000*>& GlobalDeviceList() {
 }
 
-bool devEK9000::debugEnabled = false;
+bool devEK9000::debugEnabled = true;
```
Intentional switch of debug mode on?

JJL772 replied:
Not intentional at all- good catch!

klauer reviewed (approved):
I think this support is a nice improvement 👍 

The device support logic changes look good (admittedly though my C++ is also rusty these days so it took me a bit to read). I didn't spot any real issues with the additional template/substitution files.

JJL772 committed:
FIX: remove PINI from output records

JJL772 committed:
STY: apply clang-format

klauer commented:
Checked the recent commits and looks good to me 👍 
A couple of failing builds on CI though that I didn't dig into, though. Want to check those out before we merge?

JJL772 committed:
FIX: disable motor module in CI temporarily

Mingw/windows build is broken for it, and we don't need it until motor support is PRed

JJL772 commented:
I looked at it yesterday, seems to be unrelated to any of my changes. Motor module recently updated all device support submodules, I think they broke builds in one of those device support modules.

For now I just disabled it, I'll re-enable it on my motor support branch.
